### DISCUSSION TITLE
Why there always showing the element was not present after ..... second ??? Any idea..?

### DISCUSSION THREAD

**Discussion by Leeegod** [seleniumbase/SeleniumBase#2560] [Q&A] [answered]
with SB(uc=True) as sb:
                openpage(sb)
                try:
                   versify(sb)
                except Exception:
                        openpage(sb)
                        # verify(sb)
                sb.sleep(2)
                sb.send_keys("#mat-input-0", username)
                sb.send_keys("#mat-input-1", passoword)
                sb.sleep(1.5)
                sb.driver.uc_click('span:contains(" Sign In ")')
                sb.sleep(25)
                sb.driver.uc_click("#mat-checkbox-1-input")
                sb.driver.uc_click("#mat-checkbox-2-input")
                sb.scroll_to_top()
                sb.sleep(15)
                sb.driver.uc_click('button[class="mat-focus-indicator"] + span')
                
                
                this is the code where all work propers but the last button doesn't perform click its showing error can''t found element 
                
                
                the element of the button is:
                <button mat-raised-button="" class="mat-focus-indicator btn mat-btn-lg btn-brand-orange d-none d-lg-inline-block position-absolute top-n3 right-0 z-index-999 mat-raised-button mat-button-base">
<span class="mat-button-wrapper"> Start New Booking </span>
<span matripple="" class="mat-ripple mat-button-ripple"></span>
<span class="mat-button-focus-overlay"></span></button>

Error is:

 Message:
 Element {button[class="mat-focus-indicator"] + span} was not present after 7 seconds!

is there any issue or any think i'm not getting ???

**Comment by mdmintz** [comment, marked as answer]
If the selector that you’re looking for isn’t on the page, you’ll get an error message that the element is not present. The selector that you used isn’t correct. Learn how to properly construct CSS Selectors: https://flukeout.github.io/

**Reply by Leeegod** [reply]
For this

button mat-raised-button="" class="mat-focus-indicator btn mat-btn-lg btn-brand-orange d-none d-lg-inline-block position-absolute top-n3 right-0 z-index-999 mat-raised-button mat-button-base">
span class="mat-button-wrapper"> Start New Booking </span
</span>
</buttn 

isn't this a correct way???
sb.driver.uc_click('button[class="mat-focus-indicator"] + span')

**Comment by Leeegod** [comment]
go with xpath it would work perfectly

**Comment by cozzbie** [comment]
Similar concern.

```
sb.wait_for_selector('input[type="text"][aria-label="Your name"]')
```

Works fine. SB fails by calling...

```
sb.type('input[type="text"][aria-label="Your name"]', 'Method Man')
```

...right after it. `input[type="text"][aria-label="Your name"]` is a valid CSS selector from all indications that SB seems to struggle with.

This is when attempting to join a google meeting.

Cc @mdmintz
